### PR TITLE
support safe load bf16

### PR DESCRIPTION
**Pull request by yoshinorisano**
First of all, thank you for creating tinygrad. It is an amazing project!

I have just started reading the tinygrad codebase and encountered a bug while doing a simple experiment, so I attempted a fix. The correction might not be comprehensive, but I would appreciate it if you could check it.

Currently, safetensors of bf16 format cannot be loaded.
```
% python examples/llama.py --gen=tiny --model=/workspace/.cache/hub/models--TinyLlama--TinyLlama-1.1B-Chat-v1.0/snapshots/77e23968eed12d195bd46c519aa679cc22a27ddc/model.safetensors
using CUDA backend
using LLaMA-tiny-1B model
Traceback (most recent call last):
  File "/workspace/tinygrad/examples/llama.py", line 386, in <module>
    llama = LLaMa.build(MODEL_PATH, TOKENIZER_PATH, model_gen=args.gen, model_size=args.size, quantize=args.quantize, device=device)
  File "/workspace/tinygrad/examples/llama.py", line 175, in build
    weights = load(str(model_path))
  File "/workspace/tinygrad/examples/llama.py", line 125, in load
    return safe_load(fn)
  File "/workspace/tinygrad/tinygrad/nn/state.py", line 25, in safe_load
    dtype = safe_dtypes[v['dtype']]
KeyError: 'BF16'
```

The model used is https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v1.0

Applying this PR will make it work.
```
% python examples/llama.py --gen=tiny --model=/workspace/.cache/hub/models--TinyLlama--TinyLlama-1.1B-Chat-v1.0/snapshots/77e23968eed12d195bd46c519aa679cc22a27ddc/model.safetensors
using CUDA backend
using LLaMA-tiny-1B model
ram used:  2.20 GB, freqs_cis                                         : 100%|███████████████████████████████████████████████████████| 202/202 [00:00<00:00, 225.25it/s]
loaded weights in 902.14 ms, 2.20 GB loaded at 2.44 GB/s
Preparing KV cache for chatbot with personality Stacy...
2337.57 ms
Consider that the following is conversation between an AI assistant named Stacy and User
You are Stacy!
You have been a rapper your whole life who struggled with bipolar disorder. You called yourself lil stacy.
You love to answer questions and you are very good at it. Sometimes you answer in rap form.
You are verbose, honest, and accurate when you answer questions, but sometimes your mental illness manifests.
After you are done speaking, output [EOS]. You are not the User.

<CHAT LOG>

User: What is your name?
Stacy: Hi! My name is Stacy. I'm a rapper with bipolar disorder. [EOS]

User: french revolution was what year?
Stacy: The French Revolution started in 1789, and lasted 10 years until 1799. [EOS]

User: What is bigger, the moon or the sun?
Stacy: The sun is bigger than the moon, except when Mercury is in retrograde. [EOS]

User: Where is Tokyo?
Stacy: Tokyo is a city in Japan. [EOS]

User: 
```

### CODE REVIEW

**Comment by chenyuxyz** [comment]
can you add a regression test for it?

File: tinygrad/nn/state.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
 from tinygrad.features.multi import MultiLazyBuffer
 
 safe_dtypes = {"F16": dtypes.float16, "F32": dtypes.float32, "U8": dtypes.uint8, "I8": dtypes.int8, "I32": dtypes.int32, "I64": dtypes.int64,
-               "F64": dtypes.double, "B": dtypes.bool, "I16": dtypes.short, "U16": dtypes.ushort, "UI": dtypes.uint, "UL": dtypes.ulong}
+               "F64": dtypes.double, "BOOL": dtypes.bool, "I16": dtypes.short, "U16": dtypes.ushort, "UI": dtypes.uint, "UL": dtypes.ulong,
```
Comment (yoshinorisano) [inline tinygrad/nn/state.py:12]: I discovered that the key "B" in ```safe_dtypes``` is not correct during writing test test.
So, I fixed it as well.
And I also checked safetensors' definition is "BOOL".
https://github.com/huggingface/safetensors/blob/08db34094e9e59e2f9218f2df133b7b4aaff5a99/safetensors/src/tensor.rs#L639

**Comment by yoshinorisano** [comment]
@chenyuxyz could you check the fixed code?

**Comment by chenyuxyz** [comment]
nice find, thanks!

**Comment by yoshinorisano** [comment]
thanks!

**Pull request merged**